components/encryption/Cargo.toml

```diff
@@ -0,0 +1,31 @@
+[package]
+name = "encryption"
+version = "0.0.1"
+edition = "2018"
+publish = false
+
+[features]
+protobuf-codec = ["kvproto/protobuf-codec"]
+prost-codec = ["kvproto/prost-codec"]
+
+[dependencies]
+kvproto = { git = "https://github.com/pingcap/kvproto.git", default-features = false }
+rand = "0.7"
+slog = { version = "2.3", features = ["max_level_trace", "release_max_level_debug"] }
+# better to not use slog-global, but pass in the logger
+slog-global = { version = "0.1", git = "https://github.com/breeswish/slog-global.git", rev = "0e23a5baff302a9d7bccd85f8f31e43339c2f2c1" }
+tikv_alloc = { path = "../tikv_alloc" }
+failure = "0.1"
+openssl = "0.10"
+byteorder = "1.2"
+protobuf = "2.8"
+crc32fast = "1.2"
+
+# TODO need to be abstracted
+[dependencies.rocksdb]
```

Comment:
Export the types from engine_rocks crate?

Reply:
I am planning to put abstracted stuff to engine_trait.